infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/log/PodLogHandlerToEventPublisher.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.namespace.log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.ZonedDateTime;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.cache.KubernetesMachineCache;
+import org.eclipse.che.workspace.infrastructure.kubernetes.util.RuntimeEventsPublisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class is responsible for reading the logs. It is aware of machines it should follow. */
+public class PodLogHandlerToEventPublisher implements PodLogHandler {
+
+  private final RuntimeEventsPublisher eventsPublisher;
+  private final RuntimeIdentity identity;
+  private final KubernetesMachineCache machines;
+  private final Logger LOG = LoggerFactory.getLogger(this.getClass());
+
+  public PodLogHandlerToEventPublisher(
+      RuntimeEventsPublisher eventsPublisher,
+      RuntimeIdentity identity,
+      KubernetesMachineCache machines) {
+    this.eventsPublisher = eventsPublisher;
+    this.identity = identity;
+    this.machines = machines;
+  }
+
+  /**
+   * Check if given pod is in interest of this log handler.
+   *
+   * @param podName pod to check
+   * @return true if this class cares about given podName, false otherwise
+   */
+  @Override
+  public boolean matchPod(String podName) {
+    try {
+      return machines
+          .getMachines(identity)
```

Comment:
This is hitting the database in case of `JpaKubernetesMachineCache`. Could we match the name against some list of strings supplied to the constructor instead?

Reply:
fixed